### PR TITLE
feat: Fix fetching custom papersDefinitions file

### PR DESCRIPTION
**Pull request by Merkur39**
- If custom papersDefinitions file is not found, fallback on local file
- Add warning message when custom papersDefinitions file is used

### CODE REVIEW

**Comment by Merkur39** [comment]
[Flag sur fichier de conf papiers perso](https://trello.com/c/efiBq7Yv/800-flag-sur-fichier-de-conf-papiers-perso)

File: src/utils/fetchCustomPaperDefinitions.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      const dataJSON = await dataBin.json()
+
+      return { paperConfigFilenameCustom, appFolderPath, data: dataJSON }
+    }
+
+    return { paperConfigFilenameCustom, appFolderPath, data: null }
+  } catch (error) {
+    return { paperConfigFilenameCustom, appFolderPath, data: null }
+  }
+}
```
Comment (Crash--) [inline src/utils/fetchCustomPaperDefinitions.js:44]: fetch should only fetch, the translation should not be handled here

Comment (Merkur39) [reply src/utils/fetchCustomPaperDefinitions.js:38]: Indeed, it's corrected 👍

File: src/components/App.jsx [hunk 2]
```diff
@@ -58,13 +58,24 @@ export const App = () => {
           <>
             <BarCenter>
               <MuiCozyTheme>
-                <Typography variant="h5">{client.appMetadata.slug}</Typography>
+                <Typography variant="h5">{client.appMetadata.slug} </Typography>
```
Comment (Crash--) [inline src/components/App.jsx:61]: I don't like relying of this kind of stuff. Linter can easily remove those trailing space. Why is is needed? Can't we add a margin if really needed?

File: src/components/Contexts/PapersDefinitionsProvider.jsx [hunk 3]
```diff
@@ -4,47 +4,80 @@ import useFlag from 'cozy-flags/dist/useFlag'
...
-        const paperJSONCustom = await fetchJsonFileByName(
-          client,
-          paperConfigFilenameCustom
-        )
-        if (paperJSONCustom) {
+        const {
+          paperConfigFilenameCustom,
+          appFolderPath,
+          file
+        } = await fetchCustomPaperDefinitions(client, t)
```
Comment (Crash--) [inline src/components/Contexts/PapersDefinitionsProvider.jsx:34]: no need to pass `t` anymore?

File: src/utils/fetchContentFileToJson.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { FILES_DOCTYPE } from 'src/doctypes'
+
+export const fetchContentFileToJson = async (client, data) => {
```
Comment (Crash--) [inline src/utils/fetchContentFileToJson.js:3]: why did you rename to `data`? having `file` is more understandable

File: src/components/App.jsx (hunk 2 above)
Comment (Merkur39) [reply src/components/App.jsx]: Ce n'est pas voulu, je me rends compte que le linter ne réagit pas dans ce cas précis

File: src/components/Contexts/PapersDefinitionsProvider.jsx (hunk 3 above)
Comment (Merkur39) [reply src/components/Contexts/PapersDefinitionsProvider.jsx:34]: Cette fonction fait appelle à `getOrCreateAppFolderWithReference`, qui en a besoin:
https://github.com/cozy/mespapiers/blob/feat/customPapersDef/src/utils/fetchCustomPaperDefinitions.js#L16-L19

**Review by Crash-- — APPROVED** [review]
